### PR TITLE
add nullable fields for RoleBindingRestriction

### PR DESCRIPTION
**Pull request by sallyom**
required for this: https://github.com/openshift/cluster-config-operator/pull/32

### CODE REVIEW

File: authorization/v1/types.go [hunk 1]
```diff
@@ -460,12 +460,15 @@ type RoleBindingRestriction struct {
 // field must be non-nil.
 type RoleBindingRestrictionSpec struct {
 	// UserRestriction matches against user subjects.
+	// +nullable
```
Comment (deads2k) [inline authorization/v1/types.go]: What's the difference between this and optional?  Both?

**Comment by deads2k** [comment]
it looks plausible.  I don't really know the distinction between nullable and optional and when each makes sense

/approve
/assign @damemi

File: authorization/v1/types.go (hunk 1 above)
Comment (damemi) [reply authorization/v1/types.go]: `+nullable` has an effect, which is to add `nullable: true` to the fields in the openapi validation

`+optional` actually currently doesn't do anything. Upstream controller-tools only checks for it to remove it from descriptions in the openapi spec, and instead uses `omitempty` tags to infer if fields are really optional or required. (We forked controller-tools to invert their default behavior of assuming anything not optional was required). In some places we're still keeping it around for info purposes but it's not required

**Comment by damemi** [comment]
/lgtm

comments above re:optional/nullable

**Pull request merged**